extensions/amp-date-display/1.0/date-display.js

```diff
@@ -97,16 +97,10 @@ function getDataForTemplate(props) {
 }
 
 /**
- * @param {!JsonObject} props
+ * @param {!DateDisplayDef.Props} props
  * @return {number|undefined}
  */
-function getEpoch(props) {
-  const {
-    'datetime': datetime = '',
-    'timestampMs': timestampMs = 0,
-    'timestampSeconds': timestampSeconds = 0,
-  } = props;
-
+function getEpoch({datetime = '', timestampMs = 0, timestampSeconds = 0}) {
   let epoch;
   if (datetime.toLowerCase() === 'now') {
```

Comment:
This is interesting. The optional type does not complain here. @dvoytenko

Reply:
Object dereferencing allows null/undefined objects. It's been a long-standing Closure issue.

Reply:
Mainly it's function args and return values that are required to fully comply.